mlflow/models/evaluation/base.py

```diff
@@ -329,30 +307,29 @@ def __init__(self, data, labels, name=None, path=None, feature_names=None):
                     for i in range(num_features)
                 ]
         elif isinstance(data, supported_dataframe_types):
-            if not isinstance(labels, str):
+            if not isinstance(label_col, str):
                 raise ValueError(
-                    "If data is a Pandas DataFrame or Spark DataFrame, labels must be the "
-                    "string name of a column from `data` that contains evaluation labels"
+                    "If data is a Pandas DataFrame or Spark DataFrame, `label_col` argument must "
+                    "be the name of the column which contains evaluation labels in the `data` "
+                    "dataframe."
                 )
             if isinstance(data, spark_df_type):
                 _logger.warning(
                     f"Specified Spark DataFrame is too large for model evaluation. Only "
-                    f"the first {EvaluationDataset.SPARK_DATAFRAME_LIMIT} rows will be used."
+                    f"the first {_EvaluationDataset.SPARK_DATAFRAME_LIMIT} rows will be used."
                     "If you want evaluate on the whole spark dataframe, please manually call "
                     "`spark_dataframe.toPandas()`."
                 )
-                data = data.limit(EvaluationDataset.SPARK_DATAFRAME_LIMIT).toPandas()
+                data = data.limit(_EvaluationDataset.SPARK_DATAFRAME_LIMIT).toPandas()
 
-            self._features_data = data.drop(labels, axis=1, inplace=False)
-            self._labels_data = data[labels].to_numpy()
+            self._labels_data = data[label_col].to_numpy()
 
             if feature_names is not None:
-                raise ValueError(
-                    "If `data` argument is pandas/spark dataframe, you cannot specify the "
-                    "`feature_names` argument, instead, the column names of the input "
-                    "dataframe will be used as the feature names."
-                )
-            self._feature_names = list(self._features_data.columns)
+                self._features_data = data[list(feature_names)]
+                self._feature_names = feature_names
+            else:
+                self._features_data = data.drop(label_col, axis=1, inplace=False)
```

Comment:
do we want to do a safe copy of this df before doing the drop and reusing the hash table entry to the underlying numpy arrays? A deep copy of ``data`` prior to reassigning could prevent some weird bugs if in the future any modifications to the input df are required to be done.
WDYT?

Reply:
In my test even update data the drop returned dataframe is safe and not affected:
```
>>> import pandas as pd
>>> v1 = pd.DataFrame({'a':[2,3], 'b':[3,4]})
>>> v2 = v1.drop('b', axis=1, inplace=False)
>>> v1
   a  b
0  2  3
1  3  4
>>> v2
   a
0  2
1  3
>>> v1.a[1]=4
>>> v1
   a  b
0  2  3
1  4  4
>>> v2
   a
0  2
1  3
```

Reply:
Looks like pandas has fixed this historically bad issue. I just validated with memory address locations for pd.DataFrame -> pd.Series and pd.DataFrame -> pd.DataFrame... it generates a new memory address for the underlying numpy arrays automatically now, preventing the old issues that used to crop up years ago. This is no longer a problem in pandas.
Thanks for validating!